### DISCUSSION TITLE
Create File from template, then move it to a specified folder

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [SilentVoid13/Templater#649] [Help] [answered]
Hello, 

after a long break from touching templater I need to change up a few templates, specfically what happens when I create from them.

I have  a simple file for my internal knowledge base I am slowly building, and which has been adapting over the past year or so. Currently, I use the following as a template:

    
    ---
    creation date:<% tp.file.creation_date() %>
    modification date:<% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
    type:#knowledge #todo
    alias:
    ---
    # <% tp.file.title %>
    [Wikipedia](https://en.wikipedia.org/wiki/<%tp.file.title %><% tp.file.cursor() %>)
    
    ---
    > >> 
    
    
    ---
    ## Details
    
    
    
    ---
    # Tags
    
    
    ---
    # Mentions
    ``` dataview
    list
    FROM [[]]
    Sort file.ctime Asc
    ```
    # Wikipedia
    <iframe 
    border=true
    height=800
    width=800
    src="https://en.wikipedia.org/wiki/<%tp.file.title %>"></iframe>
    
    
    ****

And then I trigger the above template exclusively\* via the following file:
    
    <% tp.file.create_new(tp.file.find_tfile("100_Template_Knowledge"), await tp.system.prompt("Set FileName"), true) %>
    <% tp.file.move("000 Output/tp.file.title")%>
    

    
Now, the first line of that script-file works perfectly. Asks for name and populates the respective spaces where it's supposed to go.
The second very evidently does not.
I have spent the last few hours trying to get the file to be automatically moved to the folder "000 Output", where all my wiki files land when they are initially created. But I can't figure out how to get the script file to move it there. The way I've tried results in a file called "tp.file.title" within the output folder, containing the textstring "[object Object]`n`n". The intended file is created, but remains in the templates-folder where the template and script both reside.
The filder 000 Output resides in the vault's root.

How do I fix this?

Thank you,  
Sincerely,  
~Gw

**Comment by welpdx** [comment, marked as answer]
Sound similar to #318

**Reply by Gewerd-Strauss** [reply]
That was somewhat insightful, but didn't really solve the issue for me unfortunately.
Or to be precise, none of what was shown there changed the outcome in any way. ObsidianSync isn't used, and I already tried doing so witout the GH-sync active, with no different results.

The file gets created in whatever folder the note I have focused at the time resides in, and then the empty artifact-file is created at the intended destination.

**Reply by welpdx** [reply]
Is your issue similar to the one in #318? 
Yea there is no fix for that issue yet. Frank said that the new update fixed it for him but I couldn't replicate his issue so I can't check.

If your problem is different than the on in #318, can you add more info on your issue? The output I mean.

So it doesn't move the file first to the correct destination with the correct template, **then** an artifact-file is created at the default new file creation location (usually root)?

Thanks

**Reply by Gewerd-Strauss** [reply]
> So it doesn't move the file first to the correct destination with the correct template, then an artifact-file is created at the default new file creation location (usually root)?

Not quite. The file is created in the folder of the currently active note - aka whatever note has typing focus, its folder gets the templated note. That can be root, or any other folder.
After that file is (correctly) populated with the template's content, _another_ file is created. That file however is created in the destination, regardless of what folder I am in when I execute the script. 
This artifact file also does not stack, and no more artifact files are born upon further executions. Or maybe they just overwrite each other, I didn't check that as the contents are always identical.


I have created a short video, which although of questionable quality, shows what's happening.

https://user-images.githubusercontent.com/80548781/170887948-d7363b3f-5907-4a71-bd2e-0b05a8d987df.mp4

 I hope this helps clarify my issue, otherwhise I'm happy to try and give more details.
 
 
 The same video can be found under this link on my google drive under this link, if you want to check specific frames:
 https://drive.google.com/file/d/1EEjTFVkY5Hb2m8Mqxa9H6WMU52ZnAMTu/view?usp=sharing
 
 - but it is not fully processed as of me submitting this comment. It's only 3 minutes, but I don't know how long google drive needs for that. So for now, the embed here has to do.

**Reply by welpdx** [reply]
Ok here are some things I am seeing:

1) When you execute the template (is it `303_event`?) on the `Untitled`, 
the `<% tp.file.move("000 Output/" + tp.file.title)%>` is going to move the currently focused `Untitled` file and move that over to the `000 Output` folder. `tp.file.move()` is not moving the file you just created, which is what I think you meant to do.
 
I suggest that instead of creating a new file with `tp.file.create_new()` and then moving it after, I recommend using `tp.file.include()` which will get all the info from your template and populate the currently focused `Untitled` file with that information. You can also change the current focused `Untitled` file's file name with `<%* await tp.file.rename("MyNewName") %>`

2) Also, the reasons why you see `[object Object]` is because you are using `<% %>`, which is outputting the results of one of those functions.  Use `<%* %>` instead.

**Reply by welpdx** [reply]
Try this: 
```js
<%* 
var title = await tp.system.prompt("Set FileName")
await tp.file.move("000 Output/" + title)
const template = tp.file.find_tfile("100_Template_Knowledge")
tR += await tp.file.include(template) 
-%>
```

Edit: Removed redundancies

**Reply by Gewerd-Strauss** [reply]
[In direct response to the code]

This will move the populated file correctly, but will break the template's population itself - `tp.file.title`-fields are no longer populated with the entered title.
I don't know if other kinds of tp-fields are also hindered or if this is exclusively the title, but mostly because I don't have any templates so far which have vastly different fields. My assumption is however that they are also not populated correctly.

![grafik](https://user-images.githubusercontent.com/80548781/170919217-3976dc5e-ccb7-4e70-a950-beb982638e8d.png)

Additionally, no artifact files are created anymore.

---

The script lies within "100 00_CreateKnowledge", and fills in from template file "100_Template_Knowledge"(can be seen in the [original post](https://github.com/SilentVoid13/Templater/discussions/649#discussion-4105417)). Both files are located in `root/002 templates/`.

> `tp.file.move()` is not moving the file you just created, which is what I think you meant to do.

That is correct, that was my intention.  And as shown above, it works in principle, although now the template population seems broken :P

**Reply by welpdx** [reply]
> I don't know if other kinds of tp-fields are also hindered or if this is exclusively the title

I'm sorry, I forgot to mention that currently there is an exclusive problem with `tp.file.title`. See #522. I don't know why at the moment. All the other functions work fine.

I suggest replacing all `tp.file.title` in your template to `this.app.workspace.getActiveFile().basename` from obsidian api. Just a few extra characters, but it should work much better. 

Or even better, you can declare a variable as the title of the document :
```js
<%* var title  = this.app.workspace.getActiveFile().basename %>
```
then just use that variable `title` everywhere else (unless you renamed the file ofc).
```js
<% title %>
```

**Reply by Gewerd-Strauss** [reply]
> I suggest replacing all tp.file.title in your template to this.app.workspace.getActiveFile().basename from obsidian api. Just a few extra characters, but it should work much better.

Tried that, but it will insert the filename of "whatever file I am currently focussed on" into the template. Meaning if I work on File "Nintendo", and I create from template  a file "Mario Galaxy" - using the file naming modal as seen above - I get a file which

* is correctly named - in this case "Mario Galaxy", but
* its `this.app.workspace.getActiveFile().basename`-fields were replaced with the string "Nintendo" - because that is the active file upon template execution, I suppose. 
Is this resolvable? Or are we just out of luck right now because tp.file.title doesn't work? 

I did not try the other solution yet, but considering it uses the same determination method I have to assume a similar result. Will update if that is not the case. 

---
Edit: Yea same problem

**Reply by welpdx** [reply]
> if I work on File "Nintendo", and I create from template a file "Mario Galaxy" - using the file naming modal as seen above - I get a file which

It sounds like what you wanted is:
1.	Create new file
2.	Have that new file be located at “000 Output/"
3.	Open new file
4.	Add stuff from “100_Template_Knowledge” into new file
5.	Have template resolve all the code in new file

This means that, if you are working on a file named "Nintendo", when you run this script, the file Nintendo is untouched. 
(I had thought that you wanted to manipulate the currently focused file! opps!)

I think I misunderstood you sorry. Try this. 
```js
<%*
var title = await tp.system.prompt("Set FileName")
const template = tp.file.find_tfile("100_Template_Knowledge")
var newTFile = await tp.file.create_new(template, "000 Output/" + title, true)
%>
```
Edit: Formatting

**Reply by Gewerd-Strauss** [reply]
1. Create new file
    2. Have that new file be located at “000 Output/"
    3. Open new file
    4. Add stuff from “100_Template_Knowledge” into new file
    5. Have template resolve all the code in new file

Clarifications:

    1. Create new file with title "X"
    2. Have that new file be located at “%rootOfVault%/000 Output/X.md"
    3. Open file [X]
    4. Add stuff from “100_Template_Knowledge” into new file
    5. Have template resolve all the code in new file, with `tp.file.title` & equivalents/workarounds resolve to title "X"

**Reply by Gewerd-Strauss** [reply]
For this code

    <%*
    var title = await tp.system.prompt("Set FileName")
    const template = tp.file.find_tfile("100_Template_Knowledge")
    var newTFile = await tp.file.create_new(template, "000 Output/" + title, true)
    %>


Assume I am in the note "root/200 University/Semester Abroard.md"
The following happens:

1. Query for filename opens → enter filename → "Test"
2. File gets created in the folder "root/200 University/000 Output/Test.md"
3. File is populated with the template's content.
4. an "Untitled"-note is created at location "root/200 University/Untitled"
    Note that this "Untitled"-file is completely empty and is identical to pressing Ctrl+N.
5. The note "root/200 University/000 Output/Test.md" gets populated based on "root/200 University/Semester Abroard.md" - so "Semester Abroard" is inserted instead of "Test".

There are three issues here:
1. The new file's output folder is relative to the current note's folder, instead of the file being created in "root/000 Output/" as intended. Meaning if I work on "root/200 University/Semester Abroard.md", the file is created in "root/200 University/000 Output/", instead of "root/000 Output/"
2. the created file's (the one located at root/200 University/000 Output/) is resolved based upon the filename of whatever file I was active in upon calling the script. Refer to [this comment and welpdx's answer](https://github.com/SilentVoid13/Templater/discussions/649#discussioncomment-2849866) for what I mean by that. - Although not resolved, I think welpdx has understood what I intend.
3. An "Untitled"-file, this time without any contents (equivalent to pressing Ctrl+N) is created.

**Reply by welpdx** [reply]
Ok I see the problem. Try this:
```js
<%* 
var rootFolders = this.app.vault.root.children
function getOutputFolder(folderName) {
    for (var i = 0; i < rootFolders.length; i++) {
        if (rootFolders[i].path == folderName) {
            return rootFolders[i]
        }
    }
}
var folder = getOutputFolder("000 Output")
const template = tp.file.find_tfile("100_Template_Knowledge")

var title = await tp.system.prompt("Set FileName")
var newTFile = await tp.file.create_new(template, title, true, folder)
%>
```
![demo](https://i.imgur.com/bxE8zGl.gif)

**Reply by Gewerd-Strauss** [reply]
That solved almost all, except that it creates empty untitled files. 
These untitled files are completely blank and do not stack, and hence "Untitled 1", "Untitled 2", ... are created.
To be precise, the templated file is created in the correct folder, then opened. Immediately afterwards, a blank "Untitled"-file is created in whatever folder the initial note was in, and focus changes onto that file instead. 



Aside from that it's perfect. It creates files correctly within the correct folder and populates the templates correctly as well.


---

Worst case I set up a watcher script with autohotkey to purge "Untitled"-files ¯\\\_(ツ)\_/¯

**Reply by welpdx** [reply]
I'm pleased to hear that it's better. As for your other issue, as you can see from the demo that a blank `untitled` file is not created for me. 
1. Could it be some templater script running from a Folder Template? Because we are creating a new file here and it will trigger any folder templates you have added in settings. 
2. Could it also be some templater function in your template that is run after the creation of this new file?

3. If not, lemme know and I'll have another script in the works for ya. Possible issue with [`tp.file.create_new()`](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#:~:text=warning%3A%20if%20you%20use%20this%20option%2C%20since%20commands%20are%20executed%20asynchronously%2C%20the%20file%20can%20be%20opened%20first%20and%20then%20other%20commands%20are%20appended%20to%20that%20new%20file%20and%20not%20the%20previous%20file.). But it'll require some more hacky coding lol

**Reply by Gewerd-Strauss** [reply]
> 1. Could it be some templater script running from a Folder Template? Because we are creating a new file here and it will trigger any folder templates you have added in settings.

I checked the templater settings, and I don't have designated folder templates enabled to begin with.

> 2. Could it also be some templater function in your template that is run after the creation of this new file?

I am not entirely sure what you mean by that if I am being honest. Within the script file I only have the contents you provided:
![grafik](https://user-images.githubusercontent.com/80548781/171098372-ebf47476-554d-4227-8662-4cc85810d671.png)
My template is still/again identical to the one in my original post.

Template Hotkeys, Folder Templates, Startup Templates, User Script Functions and User System Command Functions are all either disabled or undefined, if that is not possible. Out of curiosity I also disabled every single autohotkey script I have running, and that didn't change anything either - I didn't expect it to because in that language at least I know pretty well what I am doing, but hey. 

---

Okay, I also checked it within a blank vault - literally all I did was create an identical template and output folder and copy in my hotkey settings, everything else is pure stock - and I still generate "Untitled" to "Untitled N".

Which makes this _very_ weird. 

Unless for some reason having one vault open influences the environment of another vault opening, which would be absolutely wild.

**Reply by welpdx** [reply]
> My template is still/again identical to the one in my original post.

That is what I meant for 2. Which I see is not the problem. And wow, you did everything and the issue still persists. Weird.

Ok. try this: 
```js
<%* 
// Better way to find folder. Can find any folder in vault
const files = this.app.vault.getAllLoadedFiles()
var folder
files.forEach((file) => {
if (file.name == "000 Output") {return folder=file};
})

const template = tp.file.find_tfile("100_Template_Knowledge")
const template_content = await this.app.vault.read(template)
var title = await tp.system.prompt("Set FileName")

// from obsidian api
var newfile = await this.app.vault.create(folder.path+ "/" + title + ".md", template_content)

// Open new file. See src/core/Templater.ts#L165
const active_leaf = this.app.workspace.activeLeaf;
if (!active_leaf) {
new Notice("No active leaf", 5000);
return;
}
await active_leaf.openFile(newfile, {
state: { mode: "source" },
});
%>
```

I made this script with some obsidian api to make it do exactly what we want. 
![demo](https://i.imgur.com/njFuwu6.gif)

---
By the way, 
1) what theme is that? Looks nice!
2) how do you use ahk with obsidian o.O

**Reply by Gewerd-Strauss** [reply]
\* throws hands up in the air in frustration at how unwilling my PC is to cooperate  \*

 That's _somehow_ not working either 
(ノಠ益ಠ)ノ彡┻━┻
 It is getting ridiculous. My system seems hellbent on creating these files :P
 
 I just noticed, in the background:
 1. The Untitled file is created _before_ submitting the filename. 
 2. When pressing escape in the filename-modal, two things happen:
    - an error is thrown from templater "template parsing error, aborting. Check console for more info"
    - the untitled file that _was just created_ is deleted again. Now, if "Untitled 3" is created, also only "Untitled 3" is deleted.
 

I have attached both 
[ACompletelyBlankVault.zip](https://github.com/SilentVoid13/Templater/files/8803218/ACompletelyBlankVault.zip) - the vault you see me creating in the video - so we can test if this will happen on your machine in a vault that has had this issue on my machine, and
a [video on my google drive](https://drive.google.com/file/d/1yyb9XUtSv2n_3TXAcbnxJWJKJpAbDMme/view?usp=sharing) showing a complete fresh setup of a vault in a completely different location (heck, we're on a different partition now). 



I'm interested if anyone can replicate this issue, or if it is somehow buried within my pc. I don't usually alter a lot of settings, running windows 11 64bit right now (had to upgrade because new laptop didn't come with W10 :/ - its actually not absolute dogshit so far).


I wonder, because obsidian automatically focusses on the untitled file after creation and population of the correct one, can it delete the active (at that point then Untitled)-file? 
Obsidian has the command "Delete current file" - can that be triggered programmaticaly? Maybe with a file contents check to see if it contains any characters and then abort the deletion?


Even if this is all just a total fluke, thank you for your continued help and attempts to get this weird behaviour fixed.

---


> 1. Theme

I am using the "Shimmering Focus"-theme, albeit with some very small modifications via stylesettings. I can definetly recommend it. It's a rather minimalistic theme that is geared towards condensed information display and good readability. It is also one of the few themes in which I actually like both dark and light mode.


> 2. AHK

Not that much actually\* as of right now because I moved a lot of utility to plugins I didn't have/wasn't content with previously. But I have a variety of scripts active at all times. F.i I don't actually use obsidian's hotkeys for markdown italics/bolding/etc, because that is globally handled for me via `Lintalist.ahk` - a stupidiously useful text expander.

Aside from that I have a (not to small anymore) script to facilitate a bunch of file movements inbetween specific folders based upon custom conditions such as last edit date, tag existence and current parent folder - although I am not completely done with testing yet. The current semester kinda takes up a lot of time.

I had a custom written script for removing non-used images previously to finding a plugin doing precisely that, although I do sometimes prefer using the script because its backup capabilities fit my workflow better.
The rest is your general swath of latex helper commands, about 650 hotstrings based upon  my studies  and other fields of interest. 

Among other stuff I have a universal quicklauncher, two small GUI's for cheking ping and RAM, the ability to clear the cached ram (absurdly useful after booting up bigger minecraft modpacks, or just in general every once in a while), a small reminder script for one-off text2speech reminders (because my pc is basically running as long as I am awake either for work or leisure), a very useful screen clipper for taking partial screenshots about seven times faster than window's build-in snipping tool (and the ability to keep snippet pictures on screen as an overlay - useful when quickly referencing some small things)

And a on-screen keyboard when I train my very bad touch typing :P

Amongst maybe thirty or so small scripts build to make one task or another easier. 

---

\* At least compared to how heavily ahk-ified basically everything else is I do on my PC. I joke sometimes, but I don't think I have a single workflow left that is not in some way aided by ahk, and even if it just saves me twenty seconds - that kind amasses.


Well, and I honestly just love to tinker with it, so there's that as well.

**Reply by welpdx** [reply]
I have no issues with your vault on my computer. It does not generate any `untitled.md` files, unfortunately. 
Try this:
```js
<%* 
const files = this.app.vault.getAllLoadedFiles()
var folder
files.forEach((file) => {
if (file.name == "000 Output") {return folder=file};
})

const template = tp.file.find_tfile("100_Template_Knowledge")

await this.app.vault.read(template)
.then(async (t_content) => {
var title = await tp.system.prompt("Set FileName")
return {
title: title,
template_content: t_content
}})
.then(async (s) => {
var newfile = await this.app.vault.create(folder.path+ "/" + s.title + ".md", s.template_content);
return newfile
})
.then(async (newfile)=> {
var active_leaf = this.app.workspace.activeLeaf;
if (!active_leaf) {
new Notice("No active leaf", 5000);
} else {
await active_leaf.openFile(newfile, {
state: { mode: "source" },
});
}})
%>
```
It is the exact same script as before but I made sure that the await functions execute in sequence. The functions above line 11, couldn't have generated any new files. But the await functions below that line could. Thus, in this script, I chained the await functions to make sure that everything executes synchronously.

If that doesn't work. Then I guess it is something outside the scope of this script. Maybe it would be a good idea to reinstall obsidian. By the way, I use Obsidian V0.14.12 (Insider build). But I don't think V0.14.6 has this issue. 

Also, it is `Ctrl + Shift + I` to bring up console. 

Also, you can get rid of those annoying `LF will be replaced by CRLF` messages from git by running this `git config core.autocrlf true` in cmd console or git bash. From [SO](https://stackoverflow.com/a/5834094/14451841). 


Also, I use ahk as well. But not too proficient in it lol. I'll check your ahk repos sometime

**Reply by Gewerd-Strauss** [reply]
I haven't pushed this solution to GH yet, mostly because I am not happy with the structure for something I publish - there's a bit of "let's make this at least somewhat understandable for others" to be done :P

And a lot of my projects usually don't end up on GH either because they are in some way tied to information I cannot release, and without that I might just cut functionality by 60-80%  - and that makes very little sense to publish imo.

---

I will attempt a reinstallation the coming week. 

Running the "git config"-command in git bash didn't resolve the messages sadly. I can repeat my steps the following days to provide more details than "didn't work", but rn I gotta pack and catch a train connection :/

**Reply by welpdx** [reply]
Hey no problem. Thanks. 
By the way, did the new script fish the `untitled` issue?

**Reply by Gewerd-Strauss** [reply]
Nah, unfortunately it's the same problem.
My ahk-workaround works flawlessly so far at least, so I am rather happy about that :P

**Reply by Gewerd-Strauss** [reply]
> Try this:

Heya, short question: I tried adopting this template/script for other specific cases as well, wherein I want to target a different folder for the generated files, but I get template errors for destination paths which are more than 1 folder deep from root. MEaning I can point to 
`000 Output`, or `200 University`, but not to `200 University/08 Internships and Universities/00 Internships`. 
Any Ideas as to why that is the case?

I am currently using the newest code attempt, from [this](https://github.com/SilentVoid13/Templater/discussions/649#discussioncomment-2858321) reply. 

What must I do to make the destination folder interchangeable? The rest I figured out without much trouble, but for some reason any folder path deeper than one level is not possible.

**Reply by welpdx** [reply]
Yo @Gewerd-Strauss, are you pressing `Alt + N` or `Alt + E`?
Because `Alt + N` `Creates a new note from template`, where as what we want to do is `Alt + E` `Open Insert Template modal`

**Reply by Gewerd-Strauss** [reply]
Oh......

Oh _god_.

I always used "create from template"//<kbd>Alt+N</kbd>, because _intuitively_ that is what I am trying to do, right? Create a new file as per the terms of the template.

So if I first create a blank file, then trigger template via <kbd>Alt+E</kbd> to populate that newly created file?
Would that be the reason as to why it didn't appear to work on my end so far? I wonder if it creates "Untitled"-files when using <kbd>Alt+N</kbd> on your end as well? 

Because for <kbd>Alt-N</kbd> it creates them.  
If I create an empty file first and then populate it via <kbd>Alt+E </kbd>, no Untitled-file is created. 
If I just press <kbd>Alt+E </kbd> _wherever_ and trigger the script, a corrrectly populated file is created and then moved to the output folder.

Seems kinda counterintiutive if I am honest. 


In hindsight this however feels like the worst case of "How did I never specify _that one critial step_?!" ._.
Especially because while I can't say I understand most lines I should have gotten suspsicious at `this.app.vault.create()` interfering with the "create new file from template modal". Although I would personally see that as a very weird and non-intuitive interaction that should probably be accounted for - and if it is, please tell me so I can feel ashamed at not finding that :P

Unfortunately still can't point it at ≥2-deep folder locations :/

**Reply by welpdx** [reply]
Yes, `Alt + N` creates an `untitled` file on my end as well.

LOl we kinda talked passed each other :
- the prompt for `Alt + N` is the same as `Alt + E`. 
- [your use of modal](https://github.com/SilentVoid13/Templater/discussions/649#discussioncomment-2849866) [suggested to me `Alt + E` `Open Insert Template modal`](https://i.imgur.com/H6Ivw6V.jpeg). 
- in the beginning, you wanted to use `tp.file.create_new`, which is redundant if you are using  `Alt + N`. 
-  In one of my clarifications, I said  "1. Create a new file" meant via `tp.file.create_new`. But I think you thought of it as `Alt + N` lol

All of the above and the fact that I use `Alt + E` alot just lead me to assumed that is it what you are using as well. Sorry!

--- 

Btw, my understanding of  `Alt + N` is that it create a new file and insert   `100_Template_Knowledge` into this new file all in one go. What we are doing is using `Alt + E` to run js code that does what you want via obsidian and templater api. (for example, To create new file: We use `tp.file.create_new` or `this.app.vault.create`).

Now technically, we can add our long `<%* %>` script to our `100_Template_Knowledge` file. And use `Alt + N` to add this combined template to automatically created new file. But fair warning, there may be issues. 

---

Edit: added my understanding of `Alt + N` `Create new note from template`

**Reply by welpdx** [reply]
#### About "point it at ≥2-deep folder locations": 
Don't add path, just add the folder name. 
```js
const files = this.app.vault.getAllLoadedFiles()
var folder
files.forEach((file) => {
if (file.name == "000 Output") {return folder=file};
})
```

This section of the script finds all files and folders in vault, and **checks if the `name` is  `000 Output`**. This means that if you have the folde nested somewhere, the script will find it. **To make this work, just add `00 Internships` in the check section.**

I made it like this because before you showed me your vault, I did not know where `000 Output` was. [A previous script](https://github.com/SilentVoid13/Templater/discussions/649#discussioncomment-2852198) only found `000 Output` at the base of the vault. I quickly fixed that with the above script to find `000 Output` anywhere. 

--- 
#### A downside
The only downside with finding a folder via  name this is that it can only find folders with unique names. If you have a path like `00 Internships` > `00 Internships` > `00 Internships`, the script may find the first `00 Internships` folder.

#### Another script to remove downside
I've made another script that checks both path and name. 
So for folders with unique names you can just assign the name to var `folderPathOrName`. (e.g., `var folderPathOrName = "000 Output"`)
If you want to add a path, go ahead. It will work too. (e.g., `var folderPathOrName = "00 Internships/00 Internships/00 Internships"`)

<details><summary>Another script</summary>
<p>

```js
<%* 
var folderPathOrName = "00 Internships/00 Internships/00 Internships" // Change this for your use
const files = this.app.vault.getAllLoadedFiles()
var folder

var folderPath = tp.obsidian.normalizePath(folderPathOrName)
files.forEach((file) => {
if (file.path == folderPath) {return folder=file};
if (file.name == folderPathOrName) {return folder=file};
})
if (folder == undefined) {new Notice("can't find folder", 5000);return}

const template = tp.file.find_tfile("100_Template_Knowledge")
const template_content = await this.app.vault.read(template)
var title = await tp.system.prompt("Set FileName")

var newfile = await this.app.vault.create(folder.path+ "/" + title + ".md", template_content)

const active_leaf = this.app.workspace.activeLeaf;
if (!active_leaf) {new Notice("No active leaf", 5000);return;}
await active_leaf.openFile(newfile, {state: { mode: "source" },});
%>
```
**Note**: script is designed for  `Alt + E` `Open Insert Template modal`

</p>
</details>




<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/DUYPLG0.gif" alt="demo" />
</details>

**Reply by Gewerd-Strauss** [reply]
Alright, it has to be said:

Thank you for your continued, multi-day help across a heaps of bullshit and rewriting this time and again for some guy who doesn't manage to get it done themselves ;) This kind of extensive help is rarer than it should be.

I will mark this as accepted now, as we have resolved the last mystery - originally I would have kept it open (or left the decision to SilentVoid13) for cross-ref once someone else would come up with symptoms similar to mine, but now there is relatively little need for that. I might edit my first post to get a bit more attention to the solution for future clueless people, but tht's about it.


Thank you.

Edit: Aah, I found a single thing that is still given issues, but I will choose to ignore that:

Cancelling the script while entering the filename still creates "Null"-files :P


I don't think I can care enough anymore. That's an edgecase I can stand dealing manually with.

**Reply by welpdx** [reply]
> Cancelling the script while entering the filename still creates "Null"-files :P


<details><summary>Script no more `Null` files</summary>
<p>

```js
<%* 
var folderPathOrName = "000 Output"
//var folderPathOrName = "00 Internships/00 Internships/00 Internships" // Change this for your use
const files = this.app.vault.getAllLoadedFiles()
var folder, title

var folderPath = tp.obsidian.normalizePath(folderPathOrName)
files.forEach((file) => {
if (file.path == folderPath) {return folder=file};
if (file.name == folderPathOrName) {return folder=file};
console.log()
})
if (folder == undefined) {new Notice("can't find folder", 5000);} else {

const template = tp.file.find_tfile("100_Template_Knowledge")
const template_content = await this.app.vault.read(template)
//var title = await tp.system.prompt("Set FileName")

if (title = await tp.system.prompt("Set FileName")) {
var newfile = await this.app.vault.create(folder.path+ "/" + title + ".md", template_content)

const active_leaf = this.app.workspace.activeLeaf;
if (!active_leaf) {new Notice("No active leaf", 5000);} else {
await active_leaf.openFile(newfile, {state: { mode: "source" },});}
}}
%>
```

The problems was return don't stop the rest from executing. And `await tp.system.prompt("Set FileName")` if exited, the rest of the script executes with `title` being empty `null`.
</p>
</details>


<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/52d2Loj.gif" alt="demo" />
</details>


---

> Thank you.

You're welcome! It is my pleasure! Also, it is issues and questions like these that inspires me for my own use too!

**Comment by dredhorse** [comment]
you can't move the file you created, you move the current note you are in with tp.file.move.

But you can create the new file directly at the right place 

<%*
let title = await tp.system.prompt("Set FileName") 
let file = "000 Output/" + title
-%>
<% tp.file.create_new(tp.file.find_tfile("100_Template_Knowledge"),file, true) %>

**Reply by welpdx** [reply]
> <% tp.file.create_new(tp.file.find_tfile("100_Template_Knowledge"),title, true) %>


I think you meant:

```js
<% tp.file.create_new(tp.file.find_tfile("100_Template_Knowledge"), file, true) %>
```
Silly mistake

But I think you got what the OP wanted.

**Reply by dredhorse** [reply]
thanks, changed it

**Reply by Gewerd-Strauss** [reply]
For this code:

    <%*  
    let title = await tp.system.prompt("Set FileName")  
    let file = "/000 Output/" + title  
    -%>  
    <% tp.file.create_new(tp.file.find_tfile("100_Template_Knowledge"),file, true) %>
    
The following happens:
1. Query for filename opens → enter filename
2. File gets created in the folder %currentFolder%/000 Output/%title%.md
3. File is populated with the template's content.

There are three issues here: 
1. The new file's output folder is relative to the current note's folder, instead of the file being created in "root/000 Output/" as intended. Meaning if I work on "root/200 University/Semester Abroard.md", the file is created in "root/200 University/000 Output/", instead of "root/000 Output/"
2. the created file's (the one located at root/200 University/000 Output/) is resolved based upon the filename of whatever file I was active in upon calling the script. Refer to [this comment and welpdx's answer](https://github.com/SilentVoid13/Templater/discussions/649#discussioncomment-2849866) for what I mean by that. - Although not resolved, I think welpdx has understood what I intend.
Meaning when I trigger the script "root/200 University/Semester Abroard.md", the created file's `tp.file.title`/equivalents are resolved to "Semester Abroard", instead of the file's title itself.
3. We once again create an "Untitled"-File at "root/200 University/Untitled". Interestingly, if I repeat this process, I create "Untitled 1", "Untitled 2" and so on, with the contents
` ´n[object Object]`, where "´n"  indicates a newline. Note that this is _similar_ to the file created in the [original post](https://github.com/SilentVoid13/Templater/discussions/649#discussion-4105417), although now we have the newline _before_ the `[object Object]`-line, and not _after_ it as initially.

**Comment by AB1908** [comment]
I can't really follow what's happening here but there are two ways of going about this. The first is to create the file directly in the correct destination. `create_new` supports a `TFolder` argument for this. The other way is to use `tp.file.move` as you've done, which supports a path string if I remember correctly.
